scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1729,6 +1729,15 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
           Type.Macro(macroSplice())
         case Ident("?") if dialect.allowQuestionMarkPlaceholder =>
           next(); atPos(in.prevTokenPos, auto)(Type.Placeholder(typeBounds()))
+        case ident: Ident
+            if (ident.value == "+" || ident.value == "-") &&
+              ahead(token.is[Underscore]) &&
+              dialect.allowPlusMinusUnderscoreIdent =>
+          autoPos {
+            accept[Ident]
+            accept[Underscore]
+            Type.Name(s"${ident.value}_")
```

Comment:
should Type.Placeholder be subclassed or supplemented with `Type.CovariantPlaceholder` and `Type.ContravariantPlaceholder` and used here? after all, `+_` is closer to a placeholder than an identifier, no?

Reply:
I thought about that. However, it's unclear if just a `Name` or `Placeholder`. 
In Scala2 parser treat it as `Name` and then if kind-projector is enabled it might be transformed into `Placeholder`.

In Scala3 it will be `Placeholder`. Are there any cons of having `Placeholder` here?

Reply:
meaning, ignore the covariance (since, according to @neko-kai, it will be inferred) and just pretend it was a simple `_`? maybe.

Reply:
Currently, scalameta parses kind-projector syntax (`+*`/`-*`/`+?`/`-?`) just as `Type.Name`.
An additional thing here that we don't have variance mods for `Type.Placeholder`.

I think the best option here would be to turn `+_`, `-_` into `Type.Placeholder` for Scala3 and keep them as `Type.Name` for `Scala2*`.

@kitbellew @tgodzik wdyt?

Reply:
> I think the best option here would be to turn `+_`, `-_` into `Type.Placeholder` for Scala3 and keep them as `Type.Name` for `Scala2*`.

sounds good

Reply:
:+1: